Typescript error / downlevel-iterators?

## Environment
<!-- Run `react-native info` in your terminal and paste its contents here. -->

Honestly, I tried to run `react-native info` but it hung? That was unexpected. It only spit this bit out first:

<details>
<summary>react-native info output until it hung</summary

```
mike@isabela:~/work/Kullki/kscoreapp-rxp (mvp3-dev *) % react-native info
info 
  React Native Environment Info:
    System:
      OS: Linux 5.1 Ubuntu 19.04 (Disco Dingo)
      CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
      Memory: 685.18 MB / 31.13 GB
      Shell: 5.0.3 - /bin/bash
    Binaries:
      Node: 12.2.0 - ~/.nvm/versions/node/v12.2.0/bin/node
      Yarn: 1.16.0 - ~/.nvm/versions/node/v12.2.0/bin/yarn
      npm: 6.9.0 - ~/.nvm/versions/node/v12.2.0/bin/npm
    SDKs:
      Android SDK:
        API Levels: 15, 16, 17, 18, 19, 21, 22, 23, 24, 25, 26, 27, 28
        Build Tools: 23.0.1, 24.0.0, 24.0.2, 25.0.0, 25.0.1, 25.0.2, 26.0.0, 26.0.3, 27.0.3, 28.0.0, 28.0.1, 28.0.2, 28.0.3, 29.0.0
        System Images: android-15 | Intel x86 Atom, android-15 | Google APIs Intel x86 Atom, android-16 | ARM EABI v7a, android-16 | Intel x86 Atom, android-16 | Google APIs Intel x86 Atom, android-17 | Intel x86 Atom, android-17 | Google APIs Intel x86 Atom, android-18 | Intel x86 Atom, android-18 | Google APIs Intel x86 Atom, android-19 | Intel x86 Atom, android-19 | Google APIs Intel x86 Atom, android-21 | Intel x86 Atom, android-21 | Google APIs Intel x86 Atom, android-22 | ARM EABI v7a, android-22 | Intel x86 Atom, android-22 | Google APIs ARM EABI v7a, android-23 | Android TV Intel x86 Atom, android-23 | Intel x86 Atom, android-23 | Google APIs ARM EABI v7a, android-23 | Google APIs Intel x86 Atom, android-24 | ARM 64 v8a, android-24 | ARM EABI v7a, android-24 | Intel x86 Atom, android-24 | Google APIs Intel x86 Atom, android-25 | Google APIs ARM 64 v8a, android-25 | Google APIs ARM EABI v7a, android-25 | Google APIs Intel x86 Atom, android-26 | Google APIs Intel x86 Atom, android-27 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-Q | Google Play Intel x86 Atom_64, chromeos-67 | Chrome OS 67
    npmPackages:
      react: ^16.8.6 => 16.8.6 
      react-native: ^0.59.8 => 0.59.8 
    npmGlobalPackages:
      react-native-cli: 2.0.1

```

</details>


## Platforms
both - but really, typescript 3.4.5 with this tsconfig:

```json
{
  "compilerOptions": {
    "declaration": false,
    "noResolve": false,
    "jsx": "react",
    "reactNamespace": "RX",
    "lib": ["dom", "es5", "scripthost", "es2015"],
    "module": "commonjs",
    "target": "es5",
    "experimentalDecorators": true,
    "sourceMap": true,
    "skipLibCheck": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "outDir": "./dist/",
    "strict": true
  },
  "exclude": ["node_modules"],
  "include": ["./src/**/*"]
}
```


## Versions
<!-- Please add the used versions/branches -->
- Android: N/A
- iOS: N/A
- react-native-netinfo: 3.1.0
- react-native: 0.59.8
- react: 16.8.6

## Description

I run tsc and get this, with the above tsconfig.json:

```
[12:39:59 AM] Starting compilation in watch mode...

node_modules/@react-native-community/netinfo/src/internal/deprecatedSubscriptions.ts:23:23 - error TS2569: Type 'Set<ChangeHandler>' is not an array type or a string type. Use compiler option '--downlevelIteration' to allow iterating of iterators.

23   for (let handler of _subscriptions) {
                         ~~~~~~~~~~~~~~

node_modules/@react-native-community/netinfo/src/internal/subscriptions.ts:23:23 - error TS2569: Type 'Set<NetInfoChangeHandler>' is not an array type or a string type. Use compiler option '--downlevelIteration' to allow iterating of iterators.

23   for (let handler of _subscriptions) {
                         ~~~~~~~~~~~~~~

[12:40:06 AM] Found 2 errors. Watching for file changes.

```


## Reproducible Demo

If you paste my tsconfig.json over the project's one in the root, then run `npm run validate:typescript` you will see it

This looked like interesting related reading: https://mariusschulz.com/blog/typescript-2-3-downlevel-iteration-for-es3-es5

I suppose I could move up to es6 as my target but it would be nice for safety's sake in a ReactXP project (that touches the web) to stay on ES5 just in case. Maybe there's an easy refactor to remove the need for this style iteration?

Or at the least you should warn that the typescript requires either ES6 or downlevel-iterators, but then you might take note of the tslib/importhelpers note in the linked article

--- a/src/internal/deprecatedSubscriptions.ts
+++ b/src/internal/deprecatedSubscriptions.ts
@@ -19,10 +19,9 @@ let _latestState: DeprecatedTypes.NetInfoData | null = null;
 let _isListening = false;
 
 function _listenerHandler(state: Types.NetInfoState): void {
-  _latestState = DeprecatedUtils.convertState(state);
-  for (let handler of _subscriptions) {
-    handler(_latestState);
-  }
+  const convertedState = DeprecatedUtils.convertState(state);
+  _latestState = convertedState;
+  _subscriptions.forEach((handler): void => handler(convertedState));
 }
 
 export function add(handler: DeprecatedTypes.ChangeHandler): void {

--- a/src/internal/subscriptions.ts
+++ b/src/internal/subscriptions.ts
@@ -20,9 +20,7 @@ let _nativeEventSubscription: NativeEventSubscription | null = null;
 
 function _listenerHandler(state: Types.NetInfoState): void {
   _latestState = state;
-  for (let handler of _subscriptions) {
-    handler(_latestState);
-  }
+  _subscriptions.forEach((handler): void => handler(state));
 }
 
 export function add(

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,10 +1,9 @@
 {
   "include": ["src/**/*.ts", "example/**/*.ts", "example/**/*.tsx"],
   "compilerOptions": {
-    "target": "es6",
-    "module": "es6",
+    "target": "es5",
+    "module": "commonjs",
     "strict": true,
-    "downlevelIteration": true,
     "moduleResolution": "node",
     "lib": ["es2015", "es2016", "esnext"],
     "jsx": "react-native"

